rust/connlib/tunnel/src/tests.rs

```diff
@@ -590,27 +599,68 @@ impl ReferenceStateMachine for ReferenceState {
                         state.client_connected_cidr_resources.insert(resource);
                     }
                     Some(_) | None => {
-                        // Depending on the DNS query type, we filter the resolved addresses.
-                        let ips_resolved_by_query = all_ips.iter().copied().filter({
-                            #[allow(clippy::wildcard_enum_match_arm)]
-                            match r_type {
-                                RecordType::A => {
-                                    &(|ip: &IpAddr| ip.is_ipv4()) as &dyn Fn(&IpAddr) -> bool
-                                }
-                                RecordType::AAAA => {
-                                    &(|ip: &IpAddr| ip.is_ipv6()) as &dyn Fn(&IpAddr) -> bool
+                        // If we are NOT connected to this resource yet, we will connect and expire existing DNS records in the process.
+                        if !state.client_dns_records.contains_key(&domain) {
+                            for ip in &all_ips {
+                                for (prev_domain, prev_ips) in state.client_dns_records.iter_mut() {
+                                    prev_ips.retain(|(prev_ip, _)| {
```

Comment:
This could result in `client_dns_records` having some elements where the `Vec` of IPs is empty, that's treated equivalent to the element not being in `client_dns_records` at all, right?